src/main/kotlin/no/nav/syfo/service/DialogmoteInnkallingVarselService.kt

```diff
@@ -242,4 +256,86 @@ class DialogmoteInnkallingVarselService(
             }
         } ?: throw MissingArgumentException("EsyfovarselHendelse mangler 'data'-felt")
     }
+
+    fun getMessageText(arbeidstakerHendelse: ArbeidstakerHendelse): String? {
+        return when (arbeidstakerHendelse.type) {
+            SM_DIALOGMOTE_INNKALT -> DITT_SYKEFRAVAER_DIALOGMOTE_INNKALLING_MESSAGE_TEXT
+            SM_DIALOGMOTE_AVLYST -> DITT_SYKEFRAVAER_DIALOGMOTE_AVLYSNING_MESSAGE_TEXT
+            SM_DIALOGMOTE_REFERAT -> DITT_SYKEFRAVAER_DIALOGMOTE_REFERAT_MESSAGE_TEXT
+            SM_DIALOGMOTE_NYTT_TID_STED -> DITT_SYKEFRAVAER_DIALOGMOTE_ENDRING_MESSAGE_TEXT
+            SM_DIALOGMOTE_LEST -> ""
+            else -> {
+                log.warn("Klarte ikke mappe varsel av type ${arbeidstakerHendelse.type} ved mapping hendelsetype til ditt sykefravar melding tekst")
+                null
+            }
+        }
+    }
+
+    fun getDittSykefravarHendelseType(arbeidstakerHendelse: ArbeidstakerHendelse): String? {
+        return when (arbeidstakerHendelse.type) {
+            SM_DIALOGMOTE_INNKALT -> DittSykefravaerHendelsetypeDialogmoteInnkalling.ESYFOVARSEL_DIALOGMOTE_INNKALT.toString()
+            SM_DIALOGMOTE_AVLYST -> DittSykefravaerHendelsetypeDialogmoteInnkalling.ESYFOVARSEL_DIALOGMOTE_AVLYST.toString()
+            SM_DIALOGMOTE_REFERAT -> DittSykefravaerHendelsetypeDialogmoteInnkalling.ESYFOVARSEL_DIALOGMOTE_REFERAT.toString()
+            SM_DIALOGMOTE_NYTT_TID_STED -> DittSykefravaerHendelsetypeDialogmoteInnkalling.ESYFOVARSEL_DIALOGMOTE_NYTT_TID_STED.toString()
+            else -> {
+                log.warn("Klarte ikke mappe varsel av type ${arbeidstakerHendelse.type} ved mapping hendelsetype til ditt sykefravar hendelsetype")
+                null
+            }
+        }
+    }
+
+    private fun sendOppgaveTilDittSykefravaer(
+        arbeidstakerHendelse: ArbeidstakerHendelse,
+        varselUuid: String,
+    ) {
+        val utsendteVarsler = senderFacade.fetchAlleUferdigstilteVarslerTilKanal(arbeidstakerHendelse.arbeidstakerFnr, Kanal.DITT_SYKEFRAVAER)
+        val melding = opprettDittSykefravaerMelding(arbeidstakerHendelse, varselUuid)
+
+        if (utsendteVarsler.isNotEmpty()) {
+            when (arbeidstakerHendelse.type) {
+                SM_DIALOGMOTE_INNKALT, SM_DIALOGMOTE_NYTT_TID_STED, SM_DIALOGMOTE_AVLYST, SM_DIALOGMOTE_REFERAT -> {
+                    senderFacade.ferdigstillDittSykefravaerVarsler(arbeidstakerHendelse) // ferdigstille seg selv
+                    senderFacade.ferdigstillDittSykefravaerVarslerAvTyper(arbeidstakerHendelse, setOf<String>(SM_DIALOGMOTE_INNKALT.name, SM_DIALOGMOTE_NYTT_TID_STED.name))
+                }
+                SM_DIALOGMOTE_LEST -> {
+                    senderFacade.ferdigstillDittSykefravaerVarsler(arbeidstakerHendelse) // ferdigstille seg selv
+                }
+                else -> { log.info("Arbeidstakerhendelse ${arbeidstakerHendelse.type} skal ikke ferdigstilles") }
+            }
+        }
```

Comment:
Hvorfor må vi gjøre dette? Jeg antar at det har noe med at man ikke skal ha flere dialogmøte-varsler på en gang, men det var litt vanskelig å skjønne.

Og hvorfor gjør vi ikke det samme på brukernotifikasjoner?

Reply:
Hva er egentlig forretningslogikken for når dialogmøte-varslene skal ferdigstilles?

Reply:
Dette er for å vise kun 1 notifikasjon av samme type på ditt sykefravær. For eksempel for å ikke å vise to endringer av møte. På brukernotifikasjoner ferdigstilles kun varsel som bruker har trykket på. Her var det tanken å ferdigstille tidligere varsel hvis det kommer ny, eller ferdigstille varsel som bruker aktivt trykket på.
Logikken her er at hvis det kommer en ny innkalling, endring, avllysning eller referat varsel, ferdigstilles det tidligere versjon av samme type(om det finnes) før varsel blir sendt ut.
Lest hendelse ferdigstiller varsel på en vanlig måte.
Jeg har refaktorert det litt nå, så dette blir slikt (melding i tilfelle "lest" hendelse er null):

` val melding = opprettDittSykefravaerMelding(arbeidstakerHendelse, varselUuid)

        if (utsendteVarsler.isNotEmpty() &&
            listOf(SM_DIALOGMOTE_INNKALT, SM_DIALOGMOTE_NYTT_TID_STED, SM_DIALOGMOTE_AVLYST, SM_DIALOGMOTE_REFERAT)
                .contains(arbeidstakerHendelse.type)
        ) {
            senderFacade.ferdigstillDittSykefravaerVarsler(arbeidstakerHendelse) // ferdigstille seg selv
            senderFacade.ferdigstillDittSykefravaerVarslerAvTyper(
                arbeidstakerHendelse,
                setOf<String>(SM_DIALOGMOTE_INNKALT.name, SM_DIALOGMOTE_NYTT_TID_STED.name),
            )
        }
        if (melding != null) {
            senderFacade.sendTilDittSykefravaer(
                arbeidstakerHendelse,
                DittSykefravaerVarsel(
                    varselUuid,
                    melding,
                ),
            )
        }
        if (arbeidstakerHendelse.type == SM_DIALOGMOTE_LEST) {
            senderFacade.ferdigstillDittSykefravaerVarsler(arbeidstakerHendelse)
        }`

Reply:
Ang brukernotifikasjoner. Slikt er det i dag, flere uleste i listen, i den rekkefølgen de dukket opp. På ditt sykefravær er tanken å vise kun den aktuelle oppgaven.
![image](https://github.com/navikt/esyfovarsel/assets/4580431/343b34e3-f9bd-453b-b1f5-54f519b9655f)